bugbug/_classification.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import numpy as np
+from imblearn.metrics import (
+    geometric_mean_score,
+    make_index_balanced_accuracy,
+    specificity_score,
+)
+from sklearn.metrics.classification import precision_recall_fscore_support
+from sklearn.utils.multiclass import unique_labels
+
+
+def classification_report_imbalanced_values(
+    y_true,
+    y_pred,
+    labels=None,
+    target_names=None,
+    sample_weight=None,
+    digits=2,
+    alpha=0.1,
+):
+    """Reimplementation of imblearn.metrics.classification_report_imbalanced
+    to have access to the raw values. Copied from version 0.4.3"""
+    print("LOCALS", locals())
+
+    if labels is None:
+        labels = unique_labels(y_true, y_pred)
+    else:
+        labels = np.asarray(labels)
+
+    last_line_heading = "avg / total"
+
+    if target_names is None:
+        target_names = ["%s" % l for l in labels]
+    name_width = max(len(cn) for cn in target_names)
+    width = max(name_width, len(last_line_heading), digits)
+
+    headers = ["pre", "rec", "spe", "f1", "geo", "iba", "sup"]
+    fmt = "%% %ds" % width  # first column: class name
+    fmt += "  "
+    fmt += " ".join(["% 9s" for _ in headers])
+    fmt += "\n"
+
+    headers = [""] + headers
+    report = fmt % tuple(headers)
+    report += "\n"
+
+    # Compute the different metrics
+    # Precision/recall/f1
+    precision, recall, f1, support = precision_recall_fscore_support(
+        y_true, y_pred, labels=labels, average=None, sample_weight=sample_weight
+    )
+    # Specificity
+    specificity = specificity_score(
+        y_true, y_pred, labels=labels, average=None, sample_weight=sample_weight
+    )
+    # Geometric mean
+    geo_mean = geometric_mean_score(
+        y_true, y_pred, labels=labels, average=None, sample_weight=sample_weight
+    )
+    # Index balanced accuracy
+    iba_gmean = make_index_balanced_accuracy(alpha=alpha, squared=True)(
+        geometric_mean_score
+    )
+    iba = iba_gmean(
+        y_true, y_pred, labels=labels, average=None, sample_weight=sample_weight
+    )
+
+    result = {"targets": {}}
+
+    for i, label in enumerate(labels):
+        result["targets"][target_names[i]] = {
```

Comment:
@marco-c Do you prefer the label or the index as the dict keys?

Reply:
The label, so it's more readable

Reply:
So it should be good, I need to check with another model than defectenhancementtask